homeassistant/components/light/zwave.py

```diff
@@ -20,6 +32,15 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     node = zwave.NETWORK.nodes[discovery_info[zwave.ATTR_NODE_ID]]
     value = node.values[discovery_info[zwave.ATTR_VALUE_ID]]
+    # Make sure that we have values for the key before converting to int
+    if (value.node.manufacturer_id.strip() and
```

Comment:
I'm about to commit a different approach, basicly I have benn looking at all devices in the pepper1 database looking at the multiswitch command class, and came up with a solution to filter on Generic device class and specific device class. That combination looks to bee the easiest way to filter stuff correctly. At command Class binary switch it will be impossible to block components, because more or less all devices has that command class whatever extra capabilities it has. So switch needs to be hidden with the costumize section of the config.